src/main/java/com/shapesecurity/functional/data/ConcatList.java

```diff
@@ -83,6 +85,18 @@ protected ConcatList(int length) {
   @NotNull
   public abstract Maybe<ConcatList<T>> update(int index, @NotNull T element);
 
+  @NotNull
+  public final T[] toArray(T[] array) {
+    @SuppressWarnings("unchecked")
+    T[] target = (array.length != this.length) ?
```

Comment:
Passing in an array with the wrong length should throw an error, not be glossed over.